api/src/main/java/com/epam/pipeline/acl/user/RoleApiService.java

```diff
@@ -19,58 +19,72 @@
 import com.epam.pipeline.controller.vo.user.RoleVO;
 import com.epam.pipeline.entity.user.ExtendedRole;
 import com.epam.pipeline.entity.user.Role;
+import com.epam.pipeline.manager.security.acl.AclMask;
+import com.epam.pipeline.manager.security.acl.AclMaskList;
 import com.epam.pipeline.manager.user.RoleManager;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PostAuthorize;
+import org.springframework.security.access.prepost.PostFilter;
 import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.stereotype.Service;
 
-import java.util.Collection;
+import java.util.ArrayList;
 import java.util.List;
 
 import static com.epam.pipeline.security.acl.AclExpressions.ADMIN_ONLY;
+import static com.epam.pipeline.security.acl.AclExpressions.OR;
 import static com.epam.pipeline.security.acl.AclExpressions.OR_USER_READER;
+import static com.epam.pipeline.security.acl.AclExpressions.USER_READ_FILTER;
 
 @Service
 public class RoleApiService {
 
     @Autowired
     private RoleManager roleManager;
 
-    @PreAuthorize(ADMIN_ONLY + OR_USER_READER)
-    public Collection<Role> loadRolesWithUsers() {
-        return roleManager.loadAllRoles(true);
+    @PostFilter(USER_READ_FILTER)
+    @AclMaskList
+    public List<Role> loadRolesWithUsers() {
+        return new ArrayList<>(roleManager.loadAllRoles(true));
```

Comment:
instead of recreating list once again (in dao class we already returning a list) let's just change signature of the corresponding methods